### PR TITLE
copr: Add vectorized month

### PR DESCRIPTION
**Pull request by FateTHarlaown**
Signed-off-by: FateTHarlaown <623239185@qq.com>
PCP #5751

###  What have you changed?
Add vectorized month

###  What is the type of the changes?

Pick one of the following and delete the others:
- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

###  Refer to a related PR or issue link (optional)
#5751

### CODE REVIEW

**Comment by FateTHarlaown** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/84

**Comment by breeswish** [comment]
@youjiali1995 @MyonKeminta PTAL, thanks!

File: components/tidb_query/src/rpn_expr/impl_time.rs [hunk 1]
```diff
@@ -51,6 +52,24 @@ pub fn week_day(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<In
...
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn month(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>> {
+    let t = match t {
+        Some(v) => v,
+        _ => return Ok(None),
+    };
+    if t.is_zero() {
+        if ctx.cfg.sql_mode.contains(SqlMode::NO_ZERO_DATE) {
```
Comment (iosmanthus) [inline components/tidb_query/src/rpn_expr/impl_time.rs]: No need to do such a check. Just return `Ok(Some(t.month() as i64))`

Comment (iosmanthus) [reply components/tidb_query/src/rpn_expr/impl_time.rs]: @FateTHarlaown Please handle the conflicts~

**Comment by FateTHarlaown** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/84

**Pull request merged**